### ISSUE TITLE
Can I change operands of an instruction?

### ISSUE THREAD

**Issue by aqjune** [numba/llvmlite#420]
Is it possible to change operands of an instruction which is already created?

For example,

```
irb = ir.IRBuilder(..)
inst = irb.add(undef, undef)
```

is it possible to change operands of inst after `irb.add`?

Thanks

**Comment by alendit** [comment]
Yes, there is `Instruction.replace_usage`. You would use it like this:

```inst.replace_usage(undef, new_value)```

Which will produce the same as if you'd called `irb.add(new_value, new_value)` in the first place.

`undef` has to be the concrete reference to the value previously used as the operand, and it's every occurence will be replaced. If you want to be able to separately replace arg1 and arg2 just create two separate values for them:

```
undef1 = ir.Constant(ir.IntType(32), ir.Undefined)
undef2 = ir.Constant(ir.IntType(32), ir.Undefined)
inst = irb.add(undef1, undef2)
inst.replace_usage(undef1, new_value)
```

There is also [`Block.replace`](https://llvmlite.readthedocs.io/en/latest/user-guide/ir/values.html#llvmlite.ir.Block.replace) to replace every usage of an instruction in the whole block.

**Comment by alendit** [comment]
In case this solves your problem, you can close the issue.
